packages/ns-build/bin/ns-build.sh

```diff
@@ -18,83 +18,100 @@ do
 done
 
 # Clean-up old builds
-rm -r .next
-rm -r standalone
-rm -r deployments
+test -d .next && rm -r .next
+test -d .ns-build && rm -r .ns-build
+test -d deployments && rm -r deployments
+
+# Prepare build directory
+mkdir -p .ns-build
+
+# Hook for cleaning up failed builds.
+function cleanup {
+  # Clean up interrupted code injection build.
+  if [ -d .ns-build/app-backup ] ; then
+    # Try to be safe in preserving the original app directory.
+    # However, note the lack of error checking.
+    test -d .ns-build/app-injected && rm -r .ns-build/app-injected
+    mv app .ns-build/app-injected
+    test -d app && rm -r app
+    mv .ns-build/app-backup app
+  fi
 
-# Install necessary packages
-npm i -D serverless@3.38.0 serverless-esbuild@1.49.0 esbuild@0.19.7 serverless-http@3.2.0 ns-img-opt@1.0.2 ns-img-rdr@1.0.2
+  # Remove installed node modules
+  npm uninstall serverless serverless-esbuild esbuild serverless-http ns-img-opt ns-img-rdr
 
-# Inject code in build, and cleanup
-cp -a ./app ./app-backup
-find ./app -type f -name 'page.tsx' -exec sh -c 'printf "\nexport const runtime = '\''edge'\'';\n" >> "$0"' {} \;
-set -e
-npm run build
-set +e
-rm -r ./app
-mv ./app-backup ./app
+  test -d .next && rm -r .next
+
+  # Keep the '.ns-build' and 'deployments' directories.
+  #   'deployments' contains the generated sources,
+  #   '.ns-build' contains the back-ups for bad previous cleanup attempts.
+  # Some parts of the .ns-build, though, do not need to stick around.
+  test -d .ns-build/nodejs && rm -r .ns-build/nodejs
+}
+trap cleanup EXIT INT HUP TERM
+
+# Install necessary packages
+npm i -D \
+  serverless@3.38.0 serverless-esbuild@1.49.0 esbuild@0.19.7 serverless-http@3.2.0 ns-img-opt@1.0.2 ns-img-rdr@1.0.2 \
+  || exit 1
+
+# Inject code in build
+if [ -d app ] ; then
+  # Note that cleanup from this happens if something fails.
+  cp -a app .ns-build/app-backup || exit 1
+  find app -type f -name 'page.tsx' -exec sh -c 'printf "\nexport const runtime = '\''edge'\'';\n" >> "$0"' {} \;
+fi
+npm run build || exit 1
 
-# Keep necessary files
-cp -a .next/static .next/standalone/.next
-cp -a .next/standalone standalone
-rm standalone/server.js
-cp node_modules/ns-build/server.js standalone
-cp next.config.js standalone
 
 # Prepare deployment
-mkdir deployments
-mkdir standalone
-mkdir nodejs
+mkdir -p deployments || exit 1
+mkdir -p .ns-build || exit 1
+
+# Keep necessary files
+cp -a .next/static .next/standalone/.next || exit 1
+cp -a .next/standalone .ns-build/standalone || exit 1
```

Comment:
one of the child modules expects the `standalone` folder in the root directory

please update [this too](https://github.com/emyriounis/terraform-aws-nextjs-serverless/blob/177458ff35359baca203441b38a30c3f4752a4e5/modules/static-assets-hosting/main.tf#L25), or move the folder back to the root directory

Reply:
Good catch!  I'll get that patched up.